frontend/src/app/components/SideNavbar.tsx

```diff
@@ -0,0 +1,199 @@
+"use client";
+import Image from "next/image";
+import { useState } from "react";
+
+import logo from "../../../public/images/Logo_SPLAGen1.png";
+
+import NavCard from "./NavCard";
+import cardStyle from "./NavCard.module.css";
+import styles from "./SideNavbar.module.css";
+// Props for each navigation card on sidebar
+type CardProps = {
+  iconDark: string;
+  iconLight: string;
+  navigateTo: string;
+  message: string;
+};
+
+// There are 5 modes.
+// 1. Counselor
+// 2. Admin
+// 3. Onboarding
+// 4. Directory
+// 5. Blank
+
+// Items for the navbar's "Couneslor" mode...
+const adminItems: CardProps[] = [
+  {
+    iconDark: "dashboard_dark.svg",
+    iconLight: "dashboard_light.svg",
+    navigateTo: "/dashboard",
+    message: "Dashboard",
+  },
+  {
+    iconDark: "discussion_dark.svg",
+    iconLight: "discussion_light.svg",
+    navigateTo: "/discussion",
+    message: "Discussion",
+  },
+  {
+    iconDark: "announcements_dark.svg",
+    iconLight: "announcements_light.svg",
+    navigateTo: "/announcements",
+    message: "Announcements",
+  },
+  {
+    iconDark: "counselors_dark.svg",
+    iconLight: "counselors_light.svg",
+    navigateTo: "/counselors",
+    message: "Counselors",
+  },
+  {
+    iconDark: "admins_dark.svg",
+    iconLight: "admins_light.svg",
+    navigateTo: "/admins",
+    message: "Admins",
+  },
+];
+
+const CounselorItems: CardProps[] = [
+  {
+    iconDark: "dashboard_dark.svg",
+    iconLight: "dashboard_light.svg",
+    navigateTo: "/dashboard",
+    message: "Dashboard",
+  },
```

Comment:
You can make variables with this 
```suggestion
  Const DashboardProps: CardProps{
    iconDark: "dashboard_dark.svg",
    iconLight: "dashboard_light.svg",
    navigateTo: "/dashboard",
    message: "Dashboard",
  },
```

Then just use the variable in Counselor Items this way you don't need to repeat in the two items.